lms/djangoapps/course_api/blocks/transformers/tests/test_extra_fields.py

```diff
@@ -0,0 +1,42 @@
+"""
+Tests for ExtraFieldsTransformer.
+"""
+from django.test import override_settings
+
+# pylint: disable=protected-access
+from openedx.core.djangoapps.content.block_structure.factory import BlockStructureFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import SampleCourseFactory
+
+from ..extra_fields import ExtraFieldsTransformer
+
+
+@override_settings(COURSE_BLOCKS_API_EXTRA_FIELDS=[('course', 'other_course_settings')])
+class TestExtraFieldsTransformer(ModuleStoreTestCase):
+    """
+    Test proper behavior for ExtraFieldsTransformer
+    """
+    shard = 4
+
+    def setUp(self):
+        super(TestExtraFieldsTransformer, self).setUp()
+        self.course_key = SampleCourseFactory.create().id
+        self.course_usage_key = self.store.make_course_usage_key(self.course_key)
+        self.block_structure = BlockStructureFactory.create_from_modulestore(self.course_usage_key, self.store)
+
+    def test_transform(self):
+        # collect phase
+        ExtraFieldsTransformer.collect(self.block_structure)
+        self.block_structure._collect_requested_xblock_fields()
+
+        # transform phase
+        ExtraFieldsTransformer().transform(
+            usage_info=None,
+            block_structure=self.block_structure,
+        )
+
+        block_data = self.block_structure.get_transformer_block_data(
+            self.course_usage_key, ExtraFieldsTransformer,
+        )
+
+        self.assertEqual(block_data.other_course_settings, dict())
```

Comment:
Can we test with a non-empty value of `other_course_settings` ?

Reply:
I assumed you meant that I should replace this with a non-empty value of `other_course_settings`, which is what I did. If that's not what you wanted, please let me know.

Reply:
Yep, that's it - thanks!